File: Python/underautomation/universal_robots/primary_interface/robot_mode_data_package_event_args.py
import typing
from underautomation.universal_robots.common.robot_modes import RobotModes
from underautomation.universal_robots.common.control_modes import ControlModes
from underautomation.universal_robots.common.package_event_args import PackageEventArgs
import clr
import os
clr.AddReference(os.path.realpath(os.path.join(os.path.dirname(__file__), "..",  'lib', 'UnderAutomation.UniversalRobots.dll')))
from UnderAutomation.UniversalRobots.PrimaryInterface import RobotModeDataPackageEventArgs as robot_mode_data_package_event_args

class RobotModeDataPackageEventArgs(PackageEventArgs):
	def __init__(self, _internal = 0):
		if(_internal == 0):
			self._instance = robot_mode_data_package_event_args()
		else:
			self._instance = _internal
	@property
	def timestamp(self) -> typing.Any:
		return self._instance.Timestamp
	@timestamp.setter
	def timestamp(self, value: typing.Any):
		self._instance.Timestamp = value
	@property
	def physical_robot_connected(self) -> bool:
		return self._instance.PhysicalRobotConnected
	@physical_robot_connected.setter
	def physical_robot_connected(self, value: bool):
		self._instance.PhysicalRobotConnected = value
	@property
	def real_robot_enabled(self) -> bool:
		return self._instance.RealRobotEnabled
	@real_robot_enabled.setter
	def real_robot_enabled(self, value: bool):
		self._instance.RealRobotEnabled = value
	@property
	def robot_power_on(self) -> bool:
		return self._instance.RobotPowerOn
	@robot_power_on.setter
	def robot_power_on(self, value: bool):
		self._instance.RobotPowerOn = value
	@property
	def emergency_stopped(self) -> bool:
		return self._instance.EmergencyStopped
	@emergency_stopped.setter
	def emergency_stopped(self, value: bool):
		self._instance.EmergencyStopped = value
	@property
	def protective_stopped(self) -> bool:
		return self._instance.ProtectiveStopped
	@protective_stopped.setter
	def protective_stopped(self, value: bool):
		self._instance.ProtectiveStopped = value
	@property
	def program_running(self) -> bool:
		return self._instance.ProgramRunning
	@program_running.setter
	def program_running(self, value: bool):
		self._instance.ProgramRunning = value
	@property
	def program_paused(self) -> bool:
		return self._instance.ProgramPaused
	@program_paused.setter
	def program_paused(self, value: bool):
		self._instance.ProgramPaused = value
	@property
	def robot_mode(self) -> RobotModes:
		return RobotModes(self._instance.RobotMode)
	@robot_mode.setter
	def robot_mode(self, value: RobotModes):
		self._instance.RobotMode = value
	@property
	def control_mode(self) -> ControlModes:
		return ControlModes(self._instance.ControlMode)
	@control_mode.setter
	def control_mode(self, value: ControlModes):
		self._instance.ControlMode = value
	@property
	def target_speed_fraction(self) -> float:
		return self._instance.TargetSpeedFraction
	@target_speed_fraction.setter
	def target_speed_fraction(self, value: float):
		self._instance.TargetSpeedFraction = value
	@property
	def speed_scaling(self) -> float:
		return self._instance.SpeedScaling
	@speed_scaling.setter
	def speed_scaling(self, value: float):
		self._instance.SpeedScaling = value
	@property
	def target_speed_fraction_limit(self) -> float:
		return self._instance.TargetSpeedFractionLimit
	@target_speed_fraction_limit.setter
	def target_speed_fraction_limit(self, value: float):
		self._instance.TargetSpeedFractionLimit = value

File: PrimaryInterface/RobotModeDataPackageEventArgs.cs
//              WARNING
// This file is an empty shell containing only public C# items.
// The internal code is hidden; to access it, you need to obtain a Source licence of the library.

using Common;

namespace PrimaryInterface {
	/// <summary>
	/// Information about current robot mode
	/// </summary>
	public class RobotModeDataPackageEventArgs : PackageEventArgs {

		/// <summary>
		/// Timespan since the robot controller has started
		/// </summary>
		[PackageDescription("Timespan since the robot controller has started")]
public TimeSpan Timestamp { get; set; }

		/// <summary>
		/// Robot is connected to its controller
		/// </summary>
		[PackageDescription("Robot is connected to its controller")]
public bool PhysicalRobotConnected { get; set; }

		/// <summary>
		/// Real robot mode active. False if robot is in simulation
		/// </summary>
		[PackageDescription("Real robot mode active. False if robot is in simulation")]
public bool RealRobotEnabled { get; set; }

		/// <summary>
		/// Robot is powered on and boot is completed. If false, you need to press "ON" button to power it on
		/// </summary>
		[PackageDescription("Robot is powered on and boot is completed. If false, you need to press \"ON\" button to power it on")]
public bool RobotPowerOn { get; set; }

		/// <summary>
		/// The button Emergency Stop is pressed
		/// </summary>
		[PackageDescription("The button Emergency Stop is pressed")]
public bool EmergencyStopped { get; set; }

		/// <summary>
		/// A stop occured due to a fault detection
		/// </summary>
		[PackageDescription("A stop occured due to a fault detection")]
public bool ProtectiveStopped { get; set; }

		/// <summary>
		/// A program is running
		/// </summary>
		[PackageDescription("A program is running")]
public bool ProgramRunning { get; set; }

		/// <summary>
		/// The running program is paused
		/// </summary>
		[PackageDescription("The running program is paused")]
public bool ProgramPaused { get; set; }

		/// <summary>
		/// Current robot running mode
		/// </summary>
		[PackageDescription("Current robot running mode")]
public RobotModes RobotMode { get; set; }

		/// <summary>
		/// Current robot control mode
		/// </summary>
		[PackageDescription("Current robot control mode")]
public ControlModes ControlMode { get; set; }

		/// <summary>
		/// Overriden speed ratio between 0 (0%) and 1 (100%)
		/// </summary>
		[PackageDescription("Overriden speed ratio between 0 (0%) and 1 (100%)")]
public double TargetSpeedFraction { get; set; }

		/// <summary>
		/// Speed scaling
		/// </summary>
		[PackageDescription("Speed scaling")]
public double SpeedScaling { get; set; }

		/// <summary>
		/// Maximum target speed fraction
		/// </summary>
		[PackageDescription("Maximum target speed fraction")]
public double TargetSpeedFractionLimit { get; set; }
	}
}
